src/zns.ts

```diff
@@ -262,7 +256,7 @@ export default class Zns extends NamingService {
         };
       }
       case 'object': {
-        source = _.clone(source) as SourceDefinition;
+        source = clone(source) as SourceDefinition;
```

Comment:
Your implementation of clone is a deep clone. We don't need that. Lets limit ourselves to shallow clone like: `source = {...source}`.